tools/nodeset_compiler/nodes.py

```diff
@@ -69,7 +69,7 @@ def __init__(self):
         self.symbolicName = None
         self.writeMask = None
         self.userWriteMask = None
-        self.references = set()
+        self.references = dict()
```

Comment:
Add a short comment to document the “special” use of dict and the motivation for it.
So future generations can understand what you did.

Reply:
Done